source/Locales/LawsonLocale/SharpElementsCoincideWithSpectralPoints.lagda

```diff
@@ -0,0 +1,761 @@
+
+We define the following predicate expressing that an element `x` has decidable
+membership in compact Scott opens.
+
+\begin{code}
+
+ open Properties 𝓓
+
+ admits-decidable-membership-in-compact-scott-opens : ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
+ admits-decidable-membership-in-compact-scott-opens x =
+  Ɐ 𝒦 ꞉ ⟨ 𝒪 Scott⦅𝓓⦆ ⟩ , is-compact-open Scott⦅𝓓⦆ 𝒦 ⇒ is-decidableₚ (x ∈ₛ 𝒦)
+
+\end{code}
+
+Every sharp element admits decidable membership in compact Scott opens.
+
+\begin{code}
+
+ sharp-implies-admits-decidable-membership-in-compact-scott-opens
+  : (x : ⟨ 𝓓 ⟩∙)
+  → (is-sharp x ⇒ admits-decidable-membership-in-compact-scott-opens x) holds
+ sharp-implies-admits-decidable-membership-in-compact-scott-opens x 𝓈𝒽 𝒦 𝕜 =
+  ∥∥-rec (holds-is-prop (is-decidableₚ (x ∈ₛ 𝒦))) (uncurry ‡) ♢
+   where
+    ♢ : is-basic Scott⦅𝓓⦆ 𝒦 (spectralᴰ-implies-directed-basisᴰ Scott⦅𝓓⦆ σᴰ) holds
+    ♢ = compact-opens-are-basic
+         Scott⦅𝓓⦆
+         (spectralᴰ-implies-directed-basisᴰ Scott⦅𝓓⦆ σᴰ)
+         𝒦
+         𝕜
+
+    lemma : (xs : List (index B𝓓)) → is-decidableₚ (x ∈ₛ βσ xs) holds
+    lemma []       = inr 𝟘-elim
+    lemma (i ∷ is) =
+     ∨-preserves-decidability pt (x ∈ₛ ↑ˢ[ βₖ i ]) (x ∈ₛ 𝜸 is) † IH
+      where
+       † : is-decidableₚ (x ∈ₛ ↑ˢ[ βₖ i ]) holds
+       † = 𝓈𝒽 (β i) (basis-is-compact i)
+
+       IH : is-decidableₚ (x ∈ₛ 𝜸 is) holds
+       IH = lemma is
+
+    ‡ : (xs : List (index B𝓓)) → βσ xs ＝ 𝒦 → is-decidableₚ (x ∈ₛ 𝒦) holds
+    ‡ xs p = transport (λ - → is-decidableₚ (x ∈ₛ -) holds) p (lemma xs)
+
+\end{code}
+
+The converse also holds meaning elements that admit decidable membership in
+compact Scott opens are _exactly_ the sharp elements.
+
+\begin{code}
+
+ admits-decidable-membership-in-compact-scott-opens-implies-is-sharp
+  : (x : ⟨ 𝓓 ⟩∙)
+  → admits-decidable-membership-in-compact-scott-opens x holds
+  → is-sharp x holds
+ admits-decidable-membership-in-compact-scott-opens-implies-is-sharp x φ c 𝕜 =
+  φ ↑ˢ[ (c , 𝕜) ] (principal-filter-is-compact₀ c 𝕜)
+
+ characterization-of-sharp-elements
+  : (x : ⟨ 𝓓 ⟩∙)
+  → (admits-decidable-membership-in-compact-scott-opens x ⇔ is-sharp x) holds
+ characterization-of-sharp-elements x = † , ‡
+  where
+   † = admits-decidable-membership-in-compact-scott-opens-implies-is-sharp x
+   ‡ = sharp-implies-admits-decidable-membership-in-compact-scott-opens x
+
+\end{code}
+
+\subsection{A corollary of the characterization}
+
+Because clopens are compact in compact frames, we can also prove that admitting
+decidable membership in Scott clopens is a necessary condition for an element of
+the domain to be sharp.
+
+We do not need this result for the main result in this module, but we note it
+down regardless as it is a potentially useful observation.
+
+\begin{code}
+
+ admits-decidable-membership-in-scott-clopens : ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
+ admits-decidable-membership-in-scott-clopens x =
+  Ɐ 𝒦 ꞉ ⟨ 𝒪 Scott⦅𝓓⦆ ⟩ , is-clopen (𝒪 Scott⦅𝓓⦆) 𝒦 ⇒ is-decidableₚ (x ∈ₛ 𝒦)
+
```

Comment:
However, you should remark that any inhabited Scott clopen is the set of all elements of the domain. So, classically, there are only two Scott clopens, the whole space and the empty set (the domain is connected).

Reply:
I will update the comments to explain this. Thanks for pointing this out @martinescardo.

Reply:
Fixed in commit 7c32bfc.